process_data.py

```diff
@@ -158,67 +185,108 @@ def check_for_missing_data(hours):
+            # Only one ID given, thus only the corresponding subset is processed
+            subset_range = range(input_subset_ids[0], input_subset_ids[0]+1)
+            print("Only one latitude subset is analyzed, with subset ID {}".format(subset_range[0]))
 
-    fixed_heights_out, height_range_ceilings_out, hours_out, integration_range_ids_out, lats_out, lons_out, nc_out, output_variables = create_and_configure_output_netcdf(
-        hours, lats, lons, output_file)
+        else:
+            raise ValueError("Requested subset ID ({}) is higher than maximal subset ID {}."
+                .format(input_subset_ids[0], (n_subsets-1)))
+    elif len(input_subset_ids) == 2:
+        if all(ids < n_subsets for ids in input_subset_ids):
+            # Starting and ending ID of subsets given, the inclusive range is processed  
+            subset_range = range(input_subset_ids[0], input_subset_ids[1]+1)
+            print("A range of latitude subsets is analyzed, including subset ID {} to {}".format(subset_range[0], subset_range[-1]))
+        else:
+            raise ValueError("One of the requested subset IDs ({}, {}) is higher than maximal subset ID {}."
+                .format(input_subset_ids[0], input_subset_ids[1], (n_subsets-1)))
+    else:
+        subset_range = range(n_subsets)
+        print("All {} subsets are analyzed".format(n_subsets))
 
-    res = initialize_result_arrays(lats, lons)
+    return subset_range
 
-    # Write data corresponding to the dimensions to the output file.
-    hours_out[:] = hours
-    lats_out[:] = lats
-    lons_out[:] = lons
 
-    height_range_ceilings_out[:] = analyzed_heights['ceilings']
-    fixed_heights_out[:] = analyzed_heights['fixed']
-    integration_range_ids_out[:] = integration_range_ids
+def process_grid_subsets(output_file, input_subset_ids):
+    """"Execute analyses on the data of the complete grid and save the processed data to a netCDF file.
+
+    Args:
+        output_file (str): Name of netCDF file to which the results are saved for the respective 
+                           subset. (including format {} placeholders)
+        input_subset_ids (list): 
+
+    """
+    ds, lons, lats, levels, hours, i_highest_level = read_raw_data(start_year, final_year)
+    check_for_missing_data(hours)
 
-    # Loop over all locations to write processed data to the output file.
-    counter = 0
-    total_iters = len(lats) * len(lons)
-    start_time = timer()
 
     # Reading the data of all grid points from the NetCDF file all at once requires a lot of memory. On the other hand,
     # reading the data of all grid points one by one takes up a lot of CPU. Therefore, the dataset is analysed in
     # pieces: the subsets are read and processed consecutively.
     n_subsets = int(np.ceil(float(len(lats)) / read_n_lats_at_once))
 
-    for i_subset in range(n_subsets):
+    # Choose subsets to be processed in this run
+    subset_range = get_subset_range_from_input(input_subset_ids, n_subsets)
```

Comment:
I still don't like that you call this funciton within process_grid_subsets. Please implement as I suggested in the email:

def process_complete_grid(output_file, process_subset_id=-1):
...

if process_subset_id == -1:  # Process all
    process_subsets = range(n_subsets)
else:
    process_subsets = [process_subset_id]    
for i_subset in process_subsets:
    i_lat0 = i_subset * read_n_lats_at_once
    if i_lat0 > len(lats) - 1:
        raise ValueError("False subset id is provided.")

Reply:
here I'm not entirely sure what to do as I basically did this ==-1 etc interpretation before and then you gave me the feedback to "take this outside this function" (in case you remember, in the first pull request #5 ) so that's why I defined a function here (seemed reasonable to me, too - as it's not relevant to the processing code and just disturbes the flow of reading)

also I implemented a starting and ending user input for the subset id to start processing 10 subsets one after another with the same program call - so this short solution does not work with that anymore, of course it would again with some adjustments. This is probably also related to the question weather to give a list of the input parameters to the processing function

Maybe I'm just misunderstanding what you mean so before I change anything we can talk this over, too?

Reply:
I mean that it's better not to call get_subset_range_from_input within process_grid_subsets.
Please call both functions in main.

Reply:
So after our talk this morning I tried to apply what you imagined for processing the user input. I'm not super happy with having multiple places to verify the user input, but I guess that's just different style preferences - I hope we can agree on this solution (or something similar) :) 

I check the start/end ids in the main, then define the range in the processing and check the subset_id < n_subsets in the subset for-loop. 3d4c70b

Reply:
I changed the interpretation of the input a bit again so now its a lot as you suggested - see commit 65905b7